examples/nft/assets/nft_approval.py

```diff
@@ -0,0 +1,99 @@
+from pyteal import *
+
+def approval_program():
+    """
+    This smart contract implements a Non Fungible Token 
+    Each NFT is an (ID, ref_data, hash) triple.
+    NFT ID is the total count of NFT's in asc.
+    ref-data is represented as "https://nft-name/<total>/nft-ref"
+    hash is an hash of the external resources data.
+
+    Commands:
+        create    Creates a new NFT. Expects two additional arguments - nft-name, nft-ref 
+        tranfer   Transfers an NFT between two accounts. Expects one additional arg - NFT_ID.
+                  Additionally, two Accounts(from, to) must also be passed to the smart contract. 
+    """
+
+    # Check to see that the application ID is not set, indicating this is a creation call.
+    # Store the creator address to global state.
+    # Set total nft count to 0
+    on_creation = Seq([
+        App.globalPut(Bytes("creator"), Txn.sender()),
+        Assert(Txn.application_args.length() == Int(0)),
+        App.globalPut(Bytes("total"), Int(0)),
+        Return(Int(1))
+    ])                  
+
+	# Checks whether the sender is creator.
+    is_creator = Txn.sender() == App.globalGet(Bytes("creator"))
+
+    # Get total count of NFT's from global storage
+    total_nft = Itob(App.globalGet(Bytes("total")))
+
+    #create ref_data for NFT (https://nft-name/<total>/nft-ref)
+    ref_data = Concat(  
+        Bytes("https://"), 
+        Txn.application_args[1], # nft-name
+        Bytes(".com/"), 
+        Itob(App.globalGet(Bytes("total"))), 
+        Bytes("/"), 
+        Txn.application_args[2] # nft-ref
+    )
```

Comment:
Sorry, I wasn't clear. We are not suppose to create ref data, user can put here whatever he likes. We want to suggest in the README that URL is a good idea. So, as a `ref_data` just use `Txn.application_args[1]`.

Reply:
:+1: